mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,476 @@
+            # Also, why? really! why?
+            if params.orthogonalize_mappers:
+                Uh, S, Vh = sparsesvd(proj, proj.shape[0])
+                proj = np.dot(Uh.T, Vh)
+            #projections.append(StaticProjectionMapper(proj=csc_matrix(proj)))
+            if params.compute_recon:
+                projections.append(StaticProjectionMapper(proj=proj, recon=proj_recon))
+            else:                
+                projections.append(StaticProjectionMapper(proj=proj))
+        '''
+        return projections
+
+    def results_postproc_fx(self, results=None):
+        # XXX Divide & Rule
+        # chunk features into subsets and do them in series
+        # initialize mappers
+        ndatasets = self.ndatasets
+        nfeatures = self.nfeatures
+        params = self.params
+        projections = [csc_matrix((nfeatures, nfeatures), dtype=params.dtype) for i in range(ndatasets)]
+        if params.compute_recon:
+            projections_recon = [csc_matrix((nfeatures, nfeatures), dtype=params.dtype) for i in range(ndatasets)]
+        # Handle each results block
+        for res in results:
+            roi_feature_ids_list = [rs_.a.roi_feature_ids for rs_ in res]
+            #res = hstack(rs)
+            for isub in range(ndatasets):
+                if params.sparse_mode == 'coo':
+                    proj = self._proj_via_coo(res.samples[isub,],
+                                                    [res.a.roi_feature_ids], nfeatures)
+                else:
+                    proj = self._proj_via_dok(res.samples[isub,],
+                                                    roi_feature_ids_list, nfeatures)
+                # Cleaning up the current subject's projections to free up memory
+                res[isub,].samples = [[]]*res.shape[1]
+                if params.compute_recon:
+                    proj_recon=proj[1]
+                    projections_recon[isub] = projections_recon[isub] + proj_recon
+                proj = proj[0]
+                projections[isub] = projections[isub] + proj
+        return projections
+
+    def results_fx(self, sl=None, dataset=None, roi_ids=None, results=None):
+        # XXX Divide & Rule
+        # chunk features into subsets and do them in series
+        # initialize mappers
+        ndatasets = self.ndatasets
+        nfeatures = self.nfeatures
+        params = self.params
+        projections = [csc_matrix((nfeatures, nfeatures), dtype=params.dtype) for i in range(ndatasets)]             
+        for rs in results:
+            for isub, irs in enumerate(rs):
+                projections[isub] = projections[isub] + irs
+        # Wrap StaticProjectionMapper around the matrices
+        if params.compute_recon:
+            projections = [StaticProjectionMapper(proj=proj, recon=proj.T) for proj in projections]
+        else:                
+            projections = [StaticProjectionMapper(proj=proj) for proj in projections]
```

Comment:
I would have coded as
```python
reconkw = {'recon': proj.T} if params.compute_recon else {}
projections = [StaticProjectionMapper(proj=proj, **reconkw) for proj in projections]
```
2 lines instead of 4 and no code duplication